### PR TITLE
Kaitie/launch a b test with user ids

### PR DESCRIPTION
**Pull request by kobryan0619**
This does two things:

- Allows 10% of users to access the new section sign up flow based on their user ID
- Creates tests for each of the four access points to the new-section-sign-up flow.

Context: Past PRs have already been merged to set-up the functionality of the new-section-sign-up flow and the new-edit-section-flow.  This is the last PR that will actually make these new flows usable to 10% of users for A/B testing.

## Testing story

- Tests were created for each entry point.
- We had tested this locally as well when developing each of these components.

## Deployment strategy

- This will not be merged until we are ready to go live with the A/B tests for new section setup

## Follow-up work

- After AB testing is done we should go back in and remove unneeded tests and props as well as improve current tests.

### CODE REVIEW

**Review by hannahbergam — APPROVED** [review]
I'd like more eyes on this because I worked so closely with you on it but YAY IT IS HAPPENING🎉

File: apps/src/templates/teacherDashboard/RosterDialog.jsx [hunk 1]
```diff
@@ -280,7 +280,7 @@ class RosterDialog extends React.Component {
           </button>
           {this.props.userId % 10 === testingUserId && (
             <button
-              id="import-button"
+              id="import-button-and-redirect"
```
Comment (bethanyaconnor) [inline apps/src/templates/teacherDashboard/RosterDialog.jsx:283]: why change the id?

File: apps/src/templates/teacherDashboard/SectionActionDropdown.jsx [hunk 2]
```diff
@@ -107,13 +107,11 @@ class SectionActionDropdown extends Component {
...
   render() {
     const {sectionData, userId} = this.props;
-    const testingUserId = -1;
+    const testingUserId = 0;
 
     return (
       <span>
         <QuickActionsCell type={'header'}>
-          {/* Note that this should be uncommented when
-          ready to launch.  Also, remove line 125 for launch */}
```
Comment (bethanyaconnor) [inline apps/src/templates/teacherDashboard/SectionActionDropdown.jsx:116]: thanks!

**Review by bethanyaconnor — APPROVED** [review]
A couple of quick questions but looks good!

File: apps/src/templates/teacherDashboard/RosterDialog.jsx (hunk 1 above)
Comment (kobryan0619) [reply apps/src/templates/teacherDashboard/RosterDialog.jsx:283]: This was in service of being able to write a test for the new button that re-directs users to the edit screen.  The `RosterDialog` component has two buttons for importing currently for the A/B test.  We wrote a test for the new button to check that the button is shown when one of the 10%-users is logged in.

Comment (bethanyaconnor) [reply apps/src/templates/teacherDashboard/RosterDialog.jsx:283]: makes sense thanks!

File: apps/test/unit/templates/teacherDashboard/AddSectionDialogTest.jsx [hunk 3]
```diff
@@ -113,12 +113,12 @@ describe('AddSectionDialog', () => {
       navigateToHrefSpy.restore();
     });
 
-    /*
```
Comment (elf-code) [inline apps/test/unit/templates/teacherDashboard/AddSectionDialogTest.jsx:116]: love the block comments for the diff view

File: apps/test/unit/templates/teacherDashboard/AddSectionDialogTest.jsx [hunk 4]
```diff
@@ -113,12 +113,12 @@ describe('AddSectionDialog', () => {
...
     it('redirects to new section setup when selecting non-student participant type', () => {
       const newSection = _.cloneDeep(defaultProps.section);
       const wrapper = shallow(
         <AddSectionDialog
           {...defaultProps}
+          userId={90}
```
Comment (elf-code) [inline apps/test/unit/templates/teacherDashboard/AddSectionDialogTest.jsx:121]: magic number alert - I would add a comment above that "anything evenly divisible by 10 would pass for this test" or "userID % 10 === 0 is in the new section flow"

File: apps/test/unit/templates/teacherDashboard/TeacherDashboardHeaderTest.js [hunk 5]
```diff
@@ -94,8 +94,10 @@ describe('TeacherDashboardHeader', () => {
     expect(dropdownLinks.at(1).contains(checkmarkIcon)).to.equal(false);
   });
 
-  it('renders button to edit section details', () => {
-    const wrapper = shallow(<TeacherDashboardHeader {...DEFAULT_PROPS} />);
+  it('renders button to old edit section details', () => {
+    const wrapper = shallow(
+      <TeacherDashboardHeader {...DEFAULT_PROPS} userId={99} />
```
Comment (elf-code) [inline apps/test/unit/templates/teacherDashboard/TeacherDashboardHeaderTest.js:99]: magic number alert - this one definitely needs the comment for the opposite "userId % 10 != 0 is in the old edit section details"

File: apps/test/unit/templates/teacherDashboard/SectionActionDropdownTest.js [hunk 6]
```diff
@@ -145,22 +145,21 @@ describe('SectionActionDropdown', () => {
...
-  //   expect(wrapper.find('.edit-section-details-link').props().href).to.equal(
-  //     expectedUrl
-  //   );
-  // });
+  it('sends selected user to the new edit page', () => {
+    const wrapper = shallow(
+      <SectionActionDropdown
+        {...DEFAULT_PROPS}
+        sectionData={sections[3]}
+        userId={1539340}
```
Comment (elf-code) [inline apps/test/unit/templates/teacherDashboard/SectionActionDropdownTest.js:153]: thoughts on using 90 for magic number consistency? Or is this one different?

**Comment by elf-code** [comment]
Hi I'm (new) on the platform team, not sure if Dayne's specifically needed. But I'll give my stamp of approval!

The code looks good, just a couple of requests for more comments about the A/B test magic.

**Review by elf-code — APPROVED** [review]

**Pull request merged**